mobile/library/common/internal_engine.cc

```diff
@@ -294,6 +294,11 @@ void InternalEngine::onDefaultNetworkChanged(NetworkType network) {
         connectivity_manager_->dnsCache()->setIpVersionToRemove(absl::nullopt);
       }
     }
+    if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.drain_pools_on_network_change")) {
+      ENVOY_LOG_EVENT(debug, "netconf_immediate_drain", "DrainAllHosts");
+      connectivity_manager_->clusterManager().drainConnections(
```

Comment:
just to confirm, are we sure onDefaultNetworkChanged() gets invoked when a VPN is connected/disconnected?  Does the NetworkType go from WLAN to WLAN (same value)?

Reply:
Very good question.  I'm not 100% sure and without an iOS device I'm also not sure how to check.  It's certainly the most plausible hook we have today but I agree it warrants testing which I'm not sure how to accomplish :-/